File: NotificationPractice/app/src/main/java/com/example/notificationpractice/MainActivity.java
package com.example.notificationpractice;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.NotificationCompat;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        createChannelNotification();
        addNotification();
    }

    private void addNotification() {
        Intent notificationIntent = new Intent(this, MainActivity.class);
        PendingIntent pendingIntent = PendingIntent.getActivity(this, 0, notificationIntent,
                PendingIntent.FLAG_UPDATE_CURRENT);

        NotificationCompat.Builder builder = new NotificationCompat.Builder(this,"1");
        builder.setSmallIcon(R.drawable.ic_launcher_foreground);
        builder.setContentTitle("Notification");
        builder.setContentText("This is notification text");
        builder.setContentIntent(pendingIntent);
        builder.setAutoCancel(true);

        NotificationManager manager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        manager.notify(0, builder.build());

    }

    private void createChannelNotification(){
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
            String name = "NOTIFICATION";
            String description = "This is notification channel";
            int importance = NotificationManager.IMPORTANCE_DEFAULT;
            NotificationChannel channel = new NotificationChannel("1", name, importance);
            channel.setDescription(description);
            NotificationManager manager = getSystemService(NotificationManager.class);
            manager.createNotificationChannel(channel);
        }
    }
}

File: BroadcastReceiverTutorial/app/src/main/java/com/example/broadcastreceivertutorial/MainActivity.kt
package com.example.broadcastreceivertutorial

import android.app.NotificationManager
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.net.wifi.WifiManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.broadcastreceivertutorial.receiver.AirplaneModeReceiver
import com.example.broadcastreceivertutorial.receiver.WifiStateReceiver

class MainActivity : AppCompatActivity() {

    lateinit var receiverAirplaneMode: AirplaneModeReceiver
    lateinit var receiverWifi: WifiStateReceiver

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        receiverAirplaneMode = AirplaneModeReceiver()
        receiverWifi = WifiStateReceiver()
        
        // check airplane mode is changed
        IntentFilter(Intent.ACTION_AIRPLANE_MODE_CHANGED).also {
            registerReceiver(receiverAirplaneMode, it)
        }

        // check wifi mode is changed
        IntentFilter(WifiManager.WIFI_STATE_CHANGED_ACTION).let {
            registerReceiver(receiverWifi, it)
            val notificationManager : NotificationManager = getSystemService(
                Context.NOTIFICATION_SERVICE
            ) as NotificationManager
            receiverWifi.createNotificationChannel(notificationManager)
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        unregisterReceiver(receiverAirplaneMode)
        unregisterReceiver(receiverWifi)
    }
}